modules/series-service-impl/src/main/java/org/opencastproject/series/endpoint/SeriesRestService.java

```diff
@@ -345,6 +350,91 @@ public Response getSeriesAccessControlListJson(@PathParam("seriesID") String ser
     return getSeriesAccessControlList(seriesID);
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/allInRangeAdministrative.json")
+  @RestQuery(
+      name = "allInRangeAdministrative",
+      description = "Internal API! Returns all series (included deleted ones!) in the given "
+          + "range 'from' (inclusive) .. 'to' (exclusive). Returns at most 'limit' many series. "
+          + "Can only be used as administrator!",
+      returnDescription = "Series in the range",
+      restParameters = {
+          @RestParameter(
+              name = "from",
+              isRequired = true,
+              description = "Start of date range (inclusive) in milliseconds "
+                  + "since 1970-01-01T00:00:00Z. Has to be >=0.",
+              type = Type.INTEGER
+          ),
+          @RestParameter(
+              name = "to",
+              isRequired = false,
+              // TODO: this shows the default value as 0 despite us not setting this value!
+              description = "End of date range (exclusive) in milliseconds "
+                  + "since 1970-01-01T00:00:00Z. Has to be > 'from'.",
+              type = Type.INTEGER
+          ),
+          @RestParameter(
+              name = "limit",
+              isRequired = true,
+              description = "Maximum number of series to be returned. Has to be >0.",
+              type = Type.INTEGER
+          ),
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "All series in the range"),
+          @RestResponse(responseCode = SC_BAD_REQUEST, description = "if the given parameters are invalid"),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "If the user is not an administrator"),
+      }
+  )
+  public Response getAllInRangeAdministrative(
+      @FormParam("from") Long from,
+      @FormParam("to") Long to,
+      @FormParam("limit") Integer limit
+  ) throws UnauthorizedException {
+    // Parameter error handling
+    if (from == null) {
+      return badRequest("Required parameter 'from' not specified");
+    }
+    if (limit == null) {
+      return badRequest("Required parameter 'limit' not specified");
+    }
+    if (from < 0) {
+      return badRequest("Parameter 'from' < 0, but it has to be >= 0");
+    }
+    if (to != null && to <= from) {
+      return badRequest("Parameter 'to' <= 'from', but that is not allowed");
+    }
+    if (limit <= 0) {
+      return badRequest("Parameter 'limit' <= 0, but it has to be > 0");
+    }
+
+    try {
+      final List<Series> series = seriesService.getAllForAdministrativeRead(
+          new Date(from),
+          Optional.ofNullable(to).map(millis -> new Date(millis)),
+          limit);
+
+      Gson gson = new Gson();
```

Comment:
Gson is thread-safe, and you can make this a class field. Constructing this every time has quite some overhead and caused problems for Opencast at several places.